frontend/src/app/store/bar/bar.effects.ts

```diff
@@ -1,6 +1,35 @@
-import { Injectable } from '@angular/core';
+import { Injectable, inject } from '@angular/core';
+import { LoginService, LoginResponse } from '@bar-manager/api';
+import { Actions, act, createEffect, ofType } from '@ngrx/effects';
+import { switchMap, map, catchError, of } from 'rxjs';
+import * as fromBarActions from './bar.actions';
+import { HttpErrorResponse } from '@angular/common/http';
 
 @Injectable()
 export class BarEffects {
-  constructor() {}
+  private actions$ = inject(Actions);
+  private loginService = inject(LoginService);
+
+  private loadBar$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(fromBarActions.loadBar),
+      switchMap(action =>
+        this.loginService.login({ 'bar-code': action.barCode }).pipe(
+          map(loginResponse => fromBarActions.loadBarSuccess({ loginResponse: loginResponse })),
+          catchError((error: HttpErrorResponse) => of(fromBarActions.loadBarFailure({ error })))
+        )
+      )
+    )
+  );
+  private addBar$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(fromBarActions.addBar),
+      switchMap(action =>
+        this.loginService.postBar({ body: action.barCreationRequest }).pipe(
```

Comment:
Jetzt wo ich mir hier deine Implementierung mit den doppelten Pipes nochmal angucke, frage ich mich ob das überhaupt notwendig ist. Kannst du die map und das catchError nicht eigentlich mit in die eigentliche Pipe ziehen?

Reply:
Maybe hatte das mal als beispiel implentierung gesehen und die kopiert. Kann sein, dass das einfach / besser geht